packages/volto/src/components/manage/BlockChooser/BlockChooserButton.jsx

```diff
@@ -135,8 +135,16 @@ const BlockChooserButton = (props) => {
               onInsertBlock={
                 onInsertBlock
                   ? (id, value) => {
-                      setAddNewBlockOpened(false);
-                      onInsertBlock(id, value);
+                      if (
+                        data['@type'] === value['@type'] &&
+                        value['@type'] === 'slate'
+                      ) {
+                        onMutateBlock(block, { ...data, ...value });
+                        setAddNewBlockOpened(false);
+                      } else {
+                        setAddNewBlockOpened(false);
```

Comment:
Please move `setAddNewBlockOpened(false);` outside of the conditional, since it's done in both branches and the order doesn't matter.